### ISSUE TITLE
Camera3D.UnprojectPosition not accurate on higher resolutions

### ISSUE THREAD

**Issue by Byteron** [godotengine/godot#55255]
### Godot version

4.0.dev

### System information

Windows 11, AMD CPU & GPU

### Issue description

Camera3D.UnprojectPosition() is not accurate with higher resolutions.

the below images how unit plates that are placed on a canvas layer where the units are, using Camera3D.UnprojectPosition.

default res
![image](https://user-images.githubusercontent.com/24433899/143008187-114dca8d-cb4c-4407-b41e-b911792230c4.png)

2560x1440
![image](https://user-images.githubusercontent.com/24433899/143008888-211b49ea-4b33-4f6b-85b4-38022c4f92ab.png)

The relevant code snippet:
```csharp
if (camera.IsPositionBehind(Position))
        {
            Hide();
        }
        else
        {
            Show();
            RectPosition = camera.UnprojectPosition(Position);
        }
```

### Steps to reproduce

1) create empty project, place a cube onto some world position, and a label onto the same world position via Camera3D.UnprojectPosition. Also add Camera3D, naturally.
2) change resolution to something bigger than 1920x1080 (at 1080 I found this both work and not work depending on being borderless or not)
3) Label should be offset.

**Comment by Calinou** [comment]
Which stretch mode are you using in the project? You most likely need to multiply/divide the coordinates by the scale factor (which you can determine by dividing the current window height with the base window height, 1080 in your case).

**Comment by Byteron** [comment]
I have stretch mode set to ignore.

**Comment by Byteron** [comment]
note that I don't use resizing.
the 1440p image is with that resolution set in the project settings, so it IS the base resolution.

**Comment by madmiraal** [comment]
Probably fixed with #57082; although it's impossible to verify without an MRP, and I'm unable to reproduce the problem with the instructions provided.

**Comment by KoBeWi** [comment]
Closing based on the above comment (and lack of activity since then). Comment if you can still reproduce in newest master.
